lib/install/management/create.go

```diff
@@ -100,10 +102,69 @@ func (d *Dispatcher) createPool(conf *config.VirtualContainerHostConfigSpec, set
 	return nil
 }
 
-func (d *Dispatcher) startAppliance(conf *config.VirtualContainerHostConfigSpec) error {
+// relocateAppliance invokes the host placement library to find a suitable host (based on host
+// metrics) to relocate (migrate) the VCH to. If the VCH's current host is suitable, the VCH
+// is not relocated.
+func (d *Dispatcher) relocateAppliance() error {
 	defer trace.End(trace.Begin("", d.op))
+	var err error
+
+	// TODO(anchal): Use ranked host policy when #7459 is merged.
+	// provider := performance.NewHostMetricsProvider(d.session)
+	// rankedPolicy := placement.NewRankedHostPolicy()
+
+	randomPolicy := placement.NewRandomHostPolicy()
+	if randomPolicy.CheckHost(d.op, d.appliance) {
+		// The current host of the appliance is suitable; no migration needed.
+		return nil
+	}
+
+	host, err := randomPolicy.RecommendHost(d.op, d.appliance)
+	if err != nil {
+		msg := "Unable to obtain recommended host: %s"
+		d.op.Warnf(msg, err)
```

Comment:
Why do you log this error here? It seems like you could just return an error and then log it where you call this function.